storage/innobase/handler/handler0alter.cc

```diff
@@ -9060,6 +9105,7 @@ innobase_enlarge_column_try(
 		ut_error;
 	case DATA_BINARY:
 	case DATA_VARCHAR:
+	case DATA_CHAR:
```

Comment:
For `DATA_MYSQL` (which is a special case of `DATA_CHAR`, with `COLLATE latin1_swedish_ci`) we will still reach `ut_error` above. Please test with that collation. I would suggest something like this:
```c++
switch (col->mtype) {
case DATA_MYSQL:
case DATA_CHAR:
	ut_ad(!user_table->not_redundant());
	break;
case DATA_BINARY:

It appears to me that we are indeed storing

Reply:
It turns out that `DATA_MYSQL` is a case when charset != latin1:
```c++
	case MYSQL_TYPE_STRING:
		if (field->binary()) {
			return(DATA_FIXBINARY);
		} else if (field->charset() == &my_charset_latin1) {
			return(DATA_CHAR);
		} else {
			return(DATA_MYSQL);
		}
```